### PR TITLE
hardware wallet misc fixes

### PR DESCRIPTION
**Pull request by acharb**
misc bug fixes / changes related to hardware wallets. Pointed out in the comments

### CODE REVIEW

File: extension/src/popup/components/hardwareConnect/LedgerSign/index.tsx [hunk 1]
```diff
@@ -134,12 +145,15 @@ export const LedgerSign = () => {
           </div>
         </div>
         <div className="LedgerSign__bottom">
-          <LedgerErrorBlock error={connectError} />
+          {isDetectBtnDirty && <LedgerErrorBlock error={connectError} />}
```
Comment (acharb) [inline extension/src/popup/components/hardwareConnect/LedgerSign/index.tsx:148]: hide the connect error before user clicks "detect"

https://stellarorg.atlassian.net/browse/WAL-388

File: extension/src/popup/components/hardwareConnect/LedgerSign/index.tsx [hunk 2]
```diff
@@ -80,12 +85,18 @@ export const LedgerSign = () => {
...
             submitFreighterTransaction({
               signedXDR: res.payload,
               networkDetails,
             }),
           );
+          if (
+            submitFreighterTransaction.fulfilled.match(submitResp) &&
+            !isSwap
+          ) {
+            dispatch(addRecentAddress({ publicKey: destination }));
```
Comment (acharb) [inline extension/src/popup/components/hardwareConnect/LedgerSign/index.tsx:98]: save recent address for hardware wallet successful send
https://stellarorg.atlassian.net/browse/WAL-385

File: extension/src/popup/views/AddAccount/connect/LedgerConnect/index.tsx [hunk 3]
```diff
@@ -49,7 +49,7 @@ export const LedgerErrorBlock = ({ error }: { error: LEDGER_ERROR }) => {
   switch (error) {
     case LEDGER_ERROR.NOT_OPEN:
       errorMessage = t(
-        "Please open the Stellar app on the device and try again.",
+        "Connect device to computer and open the Stellar app on it.",
```
Comment (acharb) [inline extension/src/popup/views/AddAccount/connect/LedgerConnect/index.tsx:52]: (part of https://stellarorg.atlassian.net/browse/WAL-388)

File: extension/src/popup/components/WarningMessages/index.tsx [hunk 4]
```diff
@@ -499,28 +500,34 @@ export const NewAssetWarning = ({
...
-    );
-
-    if (signFreighterTransaction.fulfilled.match(res)) {
-      const submitResp = await dispatch(
-        submitFreighterTransaction({
-          signedXDR: res.payload.signedTransaction,
-          networkDetails,
+    if (isHardwareWallet) {
+      await dispatch(startHwSign({ transactionXDR, shouldSubmit: true }));
+      emitMetric(METRIC_NAMES.manageAssetAddUnsafeAsset, { code, issuer });
```
Comment (acharb) [inline extension/src/popup/components/WarningMessages/index.tsx:505]: ran into this bug, where adding the trustline when the NewAssetWarning popup comes up w/ hardware wallet would throw error
(I fixed this in [this PR](https://github.com/stellar/freighter/pull/656) for the ScamAssetWarning popup, forgot to change here as well)

File: extension/src/popup/views/Swap/index.tsx [hunk 5]
```diff
@@ -20,9 +20,7 @@ import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
 
 export const Swap = () => {
```
Comment (acharb) [inline extension/src/popup/views/Swap/index.tsx:21]: other misc bug found. When loading the swap page w/ no trustlines it gets caught in a useEffect loop.

I had fixed this before in https://github.com/stellar/freighter/pull/549, but looks like somewhere around the 2.3 merge conflict release it got added back 😬 . Hopefully with the new branch process we're doing won't be an issue!

**Review by piyalbasu — APPROVED** [review]

**Pull request merged**